src/ORM/EagerLoader.php

```diff
@@ -830,13 +830,18 @@ protected function _groupKeys(BufferedStatement $statement, array $collectKeys):
         $keys = [];
         foreach (($statement->fetchAll('assoc') ?: []) as $result) {
             foreach ($collectKeys as $nestKey => $parts) {
-                // Missed joins will have null in the results.
-                if ($parts[2] === true && !isset($result[$parts[1][0]])) {
-                    continue;
-                }
                 if ($parts[2] === true) {
-                    $value = $result[$parts[1][0]];
-                    $keys[$nestKey][$parts[0]][$value] = $value;
+                    // Missed joins will have null in the results.
+                    if (!array_key_exists($parts[1][0], $result)) {
+                        continue;
+                    }
+                    // Assign empty array to avoid not found association when optional.
+                    if (empty($keys[$nestKey][$parts[0]]) && !isset($result[$parts[1][0]])) {
```

Comment:
Should this be `isset()` instead of empty()? empty() can swallow '' and 0.

Reply:
I guess it could be. I will change it right away, although I guess it will never have those values as the way the array is built, but is better to be safe.

Reply:
I did the change and some other one to avoid the presence of ```['' => null]``` on the result keys set when there's some null associations.